contracts/OgvStaking.sol

```diff
@@ -280,4 +295,14 @@ contract OgvStaking is ERC20Votes {
         ogv.transfer(user, netRewards);
         emit Reward(user, netRewards);
     }
+
+    /**
+     * @dev Change delegation for `delegator` to `delegatee`.
+     *
+     * Emits events {DelegateChanged} and {DelegateVotesChanged}.
+     */
+    function _delegate(address delegator, address delegatee) internal override {
+        hasManualDelegation[delegator] = true;
```

Comment:
On second pass, this should probably get set only when user explicitly calls `delegate`. And rather than being set here I think you will need to override [this function](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Votes.sol#L143-L145) and set it there

Reply:
I thought about that but in some cases it may cause weird behaviour if we do that, for instance:
1. Alice stakes and gets self-delegated (so the flag isn't turned to true)
2. Alice stakes again but to Bob (a different address). 
In the above case, since the flag isn't true, the contract might transfer all voting powers of Alice as well to Bob, leaving Alice with no voting power. 

I can probably think of a solution to make it work but it might make the contract more complicated

Reply:
I think that the updated code is good.

The real problem here isn't about about detecting manual changes, but about knowing if delegation has been set before or not, and this setup handles that, I think.